### DISCUSSION TITLE
Monorepo Integration

### DISCUSSION THREAD

**Discussion by TechStudent10** [tauri-apps/tauri#7368] [Q&A] [answered]
How might one intergrate a Tauri project into a monorepo structure such as Nx, Turborepo, or NPM/Yarn Workspaces?

**Comment by FabianLars** [comment, marked as answer]
well, tauri itself mostly doesn't care. The projects you listed all focus on frontend stuff and not rust (right?) in which case it doesn't really matter and comes down to how you want to do it or the standard of said monorepo tools.

All tauri itself cares about is that devPath and distDir is configured accordingly, but that again depends on your frontend stack and the monorepo.

**Reply by TechStudent10** [reply]
Correct, all the monorepo solutions focus on the front-end and not Rust. Thank you

**Reply by kkuriata** [reply]
This is kinda old topic, but I think it needs some rectification. First, what do you mean by "all the monorepo solutions focus on the front-end and not Rust"?

Monorepo is just a concept of having common code base in the same place (what you normally would keep as multiple repos, now you have inside one larger repo). All your apps and libs that share common logic gathered all together. It does not matter if this is a frontend or backend project and what language they use.

Of course, you can integrate different technologies. Nx seems to be created for JS/TS projects, but it's not. It has many plugins that can extend basic functionalities:
- Rust support: `@nxrs/cargo`
- different platforms support (Electron, mobile projects): `@nstudio/xplat`

Anyways, Tauri lets you create TypeScript projects and you create in fact client app (frontend), so I don't really understand that "Rust" argument.

**Comment by samuelcastro** [comment]
FYI: https://melvinoostendorp.nl/blog/tauri-v2-nextjs-monorepo-guide
